pkg/platform/apigateway.go

```diff
@@ -61,13 +64,39 @@ type APIGatewayScrubber struct {
 	*common.AbstractScrubber
 }
 
-func NewAPIGatewayScrubber(sensitiveFields []*regexp.Regexp, kubeClientSet kubernetes.Interface) *APIGatewayScrubber {
-	abstractScrubber := common.NewAbstractScrubber(sensitiveFields, kubeClientSet)
+func NewAPIGatewayScrubber(parentLogger logger.Logger, sensitiveFields []*regexp.Regexp, kubeClientSet kubernetes.Interface) *APIGatewayScrubber {
+	abstractScrubber := common.NewAbstractScrubber(sensitiveFields, kubeClientSet, common.ReferencePrefix, common.NuclioResourceLabelKeyApiGatewayName, SecretTypeAPIGatewayConfig, parentLogger, func(name string) bool {
+		return false
+	})
 	scrubber := &APIGatewayScrubber{abstractScrubber}
 	abstractScrubber.Scrubber = scrubber
 	return scrubber
 }
 
+// RestoreAPIGatewayConfig restores an API Gateway config from a secret, in case we're running in a kube platform
+func (s *APIGatewayScrubber) RestoreAPIGatewayConfig(ctx context.Context,
+	config *APIGatewayConfig) (*APIGatewayConfig, error) {
+
+	secretMap, err := s.GetObjectSecretMap(ctx,
+		config.Meta.Name,
+		config.Meta.Namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get api gateway secret")
+	}
+	if len(secretMap) > 0 {
+
+		// restore the api gateway config
+		restoredConfig, err := s.Restore(config, secretMap)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to restore api gateway config")
+		}
+		return restoredConfig.(*APIGatewayConfig), nil
+	}
+
+	// if we're not in kube platform, or the api gateway doesn't have a secret, just return the api gateway config
```

Comment:
API gateways do not exist in local platform, and you don't do that check in this function anyways...